### ISSUE TITLE
*Feature Request* Import Obstacles

### ISSUE THREAD

**Issue by Geryones** [mavlink/qgroundcontrol#7100]
The import of a list of obstacles via KML would be really helpful. For the beginning, it would be enough, if the obstacles could be displayed even if they do not yet affect the flightpattern. 

Select KML File with all Obstacles 
![image](https://user-images.githubusercontent.com/10724637/50397829-51694e00-0773-11e9-927e-09814a362622.png)

Display the whole file content (here power lines in blue, anything else in red)
![image](https://user-images.githubusercontent.com/10724637/50397864-883f6400-0773-11e9-94a7-0255536113a2.png)

I'll try to implement it on my own, but every bit of help is more than welcome :)

**Comment by jstrebel** [comment]
Hi, are you saying KML Import is not working at all?

**Comment by Geryones** [comment]
No, KML import is working for a single Mission Item. At least I, was unable to import multiple items at once

**Comment by DonLakeFlyer** [comment]
> No, KML import is working for a single Mission Item.

Which is a correct statement. You can't import KML which contains multiple polygons or polylines.

If you look at FlightDisplayViewMap you will see multiple examples of map overlays in there which are currently supported: AirMap, Orbit visuals and so forth. You need to add a new type from the which the data is provided by app.

**Comment by Geryones** [comment]
Thank you very much. 
I found the type CustomMapItems in it, with the comment "Allow custom builds to add map items". Maybe I can start there? 
In the MissionController, I am able to create a QList<QGCMapPolygon> for all Polygons in a kml file, but I fail when I try to visualize them. I tried to add them to _visualItems, but that resulted in QGC crashing. 

How do I get from my List of polygons, to actually displaying them? They should not be interactiv or something, just static objects rendered on the map. 
What step am I missing?

**Comment by DonLakeFlyer** [comment]
> How do I get from my List of polygons, to actually displaying them?

You write a bunch of code :) You're going to need to educate yourself on Qml. Look at how a GeoFence is displayed for examples.

**Comment by Geryones** [comment]
This much I learned allready :) 
KK will do! Thank you
